chapters/classes.tex

```diff
@@ -383,14 +383,27 @@ \subsection{Component Variability Prefixes discrete, parameter, constant}\label{
 definition equations is given in \cref{variability-of-expressions}.
 
 \begin{nonnormative}
-A discrete-time variable is a piecewise constant signal which
-changes its values only at event instants during simulation. Such types
-of variables are needed in order that special algorithms, such as the
-algorithm of Pantelides for index reduction, can be applied (it must be
-known that the time derivative of these variables is identical to zero).
-Furthermore, memory requirements can be reduced in the simulation
-environment, if it is known that a component can only change at event
-instants.
+For \lstinline!Real! variables we can distinguish two subtly different categories: discrete-time and piecewise constant.
+The discrete-time \lstinline!Real! variables must be assigned in a when-clause.
+The discrete-time variables can be used as argument to \lstinline!pre!.
+The discrete-time variables are a subset of all variables that are piecewise constant and only change at events.
```

Comment:
Possibly already incorporated in first sentence.

Reply:
Agreed

Reply:
Note that there's a subtle different in that a Real variable could be piecewise constant but change at non-event time points. I have no idea how you would construct and verify that, so I'll just ignore it.

Reply:
It sounds like it could be worth giving some thought, as it could actually add value to this non-normative text on a non-trivial topic with several subtleties.

Reply:
I made an attempt at this.

Reply:
What about a simple `if`-expression with `noEvent` condition?
```
  Real x = if noEvent(time^2 > 0.5) then 1.0 else -1.0;
```

Reply:
Yes. And preferably without the super-script as it just makes it look even more contrived than necessary. If we want more than one event an alternative is `Real x=if noEvent(cos(time) >0.5) then 1.0 else -1.0;`

Reply:
I didn't want more than one event, but an easy way to construct a jump at a time that would be unlikely to coincide with the output grid.

Reply:
I understand that, but to me avoiding events for periodic changes makes even more sense.

Reply:
Sure, `cos(time) > 0.5` would also be fine.

Reply:
I have added such a case.

Reply:
Looks good.  Resolving.